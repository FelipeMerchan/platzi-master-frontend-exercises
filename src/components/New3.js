import React from 'react'

import './styles/New3.css'

function New3() {
  return (
    <section className="New3">
      <h3>Lorem ipsum dolor sit amet, consectetuer adipiscing elit</h3>
      <div>
        <div>
          <h5 className="line-top text-center text-uppercase">Lorem ipsum </h5>
          <p className="text-justify">
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
            nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
            volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
            ullamcorper.
          </p>
        </div>
        <p className="text-justify">
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
          nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
          volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
          ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
        </p>
        <p className="text-justify">
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
          nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
          volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
          ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
        </p>
        <p className="text-justify">
          Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
          nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
          volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
          ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
        </p>
      </div>
    </section>
  )
}

export default New3