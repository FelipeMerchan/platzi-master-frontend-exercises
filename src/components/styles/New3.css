.New3 {
  padding: var(--space-2) 0;
  border-bottom: 1px solid var(--black-color);
}

.New3 > div {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-column-gap: var(--space);
}

.New3 > h3 {
  margin-bottom: var(--space);
  font-style: italic;
  font-size: 1.4rem;
}
