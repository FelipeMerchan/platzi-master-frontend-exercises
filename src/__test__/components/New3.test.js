import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import New3 from '../../components/New3'

describe('Probar el componente <New3 />', () => {
  const new3 = shallow(<New3 />)

  test('Render del componente New3',() => {
    expect(new3.length).toEqual(1)
  })
  test('Render del tÃ­tulo', () => {
    expect(new3.find('h3').text()).toEqual('Lorem ipsum dolor sit amet, consectetuer adipiscing elit')
  })
  test('Render con el className text-center', () => {
    expect(new3.find('section').hasClass('New3')).toBe(true)
  })
})

describe('New3 Snapshot', () => {
  test('Comprobar la interfaz de usuario del componente New3', () => {
    const new3 = create(<New3 />)
    expect(new3.toJSON()).toMatchSnapshot()
  })
})