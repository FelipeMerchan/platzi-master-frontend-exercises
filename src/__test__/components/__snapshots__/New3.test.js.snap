// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New3 Snapshot Comprobar la interfaz de usuario del componente New3 1`] = `
<section
  className="New3"
>
  <h3>
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit
  </h3>
  <div>
    <div>
      <h5
        className="line-top text-center text-uppercase"
      >
        Lorem ipsum 
      </h5>
      <p
        className="text-justify"
      >
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper.
      </p>
    </div>
    <p
      className="text-justify"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
    </p>
    <p
      className="text-justify"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
    </p>
    <p
      className="text-justify"
    >
      Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
    </p>
  </div>
</section>
`;
